src/storage/gcs/storage.c

```diff
@@ -98,6 +98,61 @@ struct StorageGcs
     HttpClient *authClient;                                         // Client to service auth requests
 };
 
+/***********************************************************************************************************************************
+Parse HTTP JSON response containing an authentication token and expiration
+
+Note that the function is intended to run directly in the caller's mem context and results will be placed in the caller's prior mem
+context.
```

Comment:
Isn't that somehow backwards? Shouldn't the caller decide which memory context the result ends up in?

And if this is the intended behavior, then it might be good to indicate that somehow in the function name, like naming it storageGcsAuthTokenIntoPrior() (don't particularly like that name but it is just an example).

Reply:
Yeah, we could pass in the mem context, but moving to prior is a pattern we use pretty regularly. If this function were extern'd we probably make the caller do their own move, but since it is module level I think it is fine with the documentation. Thoughts, @cmwshang?

Reply:
> Yeah, we could pass in the mem context, but moving to prior is a pattern we use pretty regularly. If this function were extern'd we probably make the caller do their own move, but since it is module level I think it is fine with the documentation. Thoughts, @cmwshang?

If it were not a static module-level function (i.e. extern'd), I would agree with Jan.